File: dongyimaicartweb/src/main/webapp/js/service/cartservice.js
app.service('cartservice',function ($http) {

    this.findUserById = function () {
        return $http.get('../address/findUserById.do')
    }


    this.addGoodsToCart = function (itemId,num) {
        return $http.post('../cart/addGoodsToCart.do?itemId='+itemId+'&num='+num);
    }

    this.findCartList = function () {
        return $http.get('../cart/findCookieCartList.do');
    }

})

File: dongyimaicartinterface/src/main/java/com/offcn/cart/service/CartService.java
package com.offcn.cart.service;

import com.offcn.pojo.Cart;

import java.util.List;

public interface CartService {

    public List<Cart> addGoodsToCartList(List<Cart> cartList, long itemId, int num);

    public void addGoodsToRedis(String name,List<Cart> cartList);

    public List<Cart> findGoodsFromRedis(String name);

    public List<Cart> mergeCartList(List<Cart> cookieList,List<Cart> redisList);

}
